src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,366 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Handles UI aspects of the duke.Duke chatbot.
+ *
+ * @author Yuvaraj Kumaresan
+ */
+public class Ui {
+
+    /**
+     * The storage used to interact with the user.
+     */
+    protected Storage storage;
+
+    /**
+     * Constructor.
+     *
+     * @param storage The storage used to get item to interact with user.
+     */
+    public Ui(Storage storage) {
+        this.storage = storage;
+    }
+
+    String divider = "--------------------------------------------------" +
+            "--------------------------------------------------------------";
+
+    /**
+     * Handles UI output for list keyword.
+     */
+    public void list() {
+        if (storage.taskList.arrayList.size() == 0) {
+            System.out.println(divider);
+            System.out.println("The list is empty, please add more items to display them here.");
+            System.out.println(divider);
+        } else {
+            System.out.println(divider);
+            System.out.println("Here are the tasks in your list : ");
+            for (int i = 0; i < storage.taskList.arrayList.size(); i++) {
+                System.out.println((i + 1) + ". " + storage.taskList.arrayList.get(i).toString());
+            }
+            System.out.println(divider);
+        }
+    }
+
+    /**
+     * Handles UI output for bye keyword.
+     *
+     * @throws IOException
+     */
+    public void bye() throws IOException {
+        System.out.println(divider);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(divider);
+        storage.writeToFile();
+        System.exit(0);
+    }
+
+    /**
+     * Handles UI output for delete keyword with no line number.
+     */
+    public void deleteNoNumber() {
```

Comment:
I was confused initially by the naming of this method, maybe you could change it to what is shown below instead ?
```suggestion
    public void deleteWithNoNumber() {
```